server/protocols/protocols-smtp/src/main/java/org/apache/james/smtpserver/priority/SmtpMtPriorityParameters.java

```diff
@@ -0,0 +1,100 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.smtpserver.priority;
+
+import static org.apache.james.queue.api.MailPrioritySupport.LOW_PRIORITY;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import org.apache.james.queue.api.MailPrioritySupport;
+import org.apache.mailet.Attribute;
+import org.apache.mailet.AttributeValue;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Preconditions;
+
+public class SmtpMtPriorityParameters {
+    public static final String MT_PRIORITY_PARAMETER = "MT-PRIORITY";
+    public static final int MIN_MT_PRIORITY_VALUE = -9;
+    public static final int MAX_MT_PRIORITY_VALUE = 9;
+
+    public static class MtPriority {
+
+        public static MtPriority of(String value) {
+            Preconditions.checkNotNull(value);
+            Integer priority = Integer.parseInt(value);
+            return new SmtpMtPriorityParameters.MtPriority(priority);
+        }
+
+        public static Optional<MtPriority> from(String value) {
+            if (Integer.parseInt(value) < MIN_MT_PRIORITY_VALUE || Integer.parseInt(value) > MAX_MT_PRIORITY_VALUE) {
+                throw new IllegalArgumentException("Invalid priority: According to RFC-6710 allowed values are from -9 to 9 inclusive");
+            }
+            return Optional.of(value)
+                .map(SmtpMtPriorityParameters.MtPriority::of);
+        }
```

Comment:
Why an optional? We actually always return a value....

Also we call `Integer.parseInt(value)` 3 times, twice in the check, once in `of`.

Suggestion:

 - [ ] Move the check (-9 < p < 9) into the constructor
 - [ ] And remove `from` method
 - [ ] Adapt callers